File: erettsegi/kozep/2026_may/python/nyomas.py
with open('nyomas.txt') as nyomas_input:
    meresek = [ int(k) for k in nyomas_input.read().strip().split(', ') ]

legkisebb_index = min((i for i in range(0, len(meresek))), key = lambda i: meresek[i])

print(f'A legkisebb mért érték: {meresek[legkisebb_index]}, sorszáma: {legkisebb_index + 1}')

bekert_ertek = int(input('Minél kisebb értékeket keres? (egész szám) '))
kisebb_ertekek_szama = sum(1 for k in meresek if k < bekert_ertek)

print(f'{bekert_ertek} alatti mérések száma: {kisebb_ertekek_szama}')

legnagyobb_csokkenes = max(k - l for k, l in zip(meresek[:-1], meresek[1:]))

print(f'A két mérés közötti legnagyobb csökkenés: {legnagyobb_csokkenes}')

File: erettsegi/kozep/2026_may/c#/Nyomas.cs
var meresek = File.ReadAllText("nyomas.txt").Trim().Split(", ")
                  .Select(int.Parse)
                  .ToArray();

var legkisebbIndex = Enumerable.Range(0, meresek.Length)
                               .MinBy(i => meresek[i]);

Console.WriteLine($"A legkisebb mért érték: {meresek[legkisebbIndex]}, sorszáma: {legkisebbIndex + 1}");
Console.Write("Minél kisebb értékeket keres? (egész szám) ");

var bekertErtek = int.Parse(Console.ReadLine()!);
var kisebbErtekekSzama = meresek.Count(k => k < bekertErtek);

Console.WriteLine($"{bekertErtek} alatti mérések száma: {kisebbErtekekSzama}");

var legnagyobbCsokkenes = Enumerable.Zip(meresek.Take(meresek.Length - 1), meresek.Skip(1))
                                    .Max((k) => k.First - k.Second);

Console.WriteLine($"A két mérés közötti legnagyobb csökkenés: {legnagyobbCsokkenes}");
